ignite/pkg/cosmosclient/txservice.go

```diff
@@ -27,36 +27,43 @@ func (s TxService) Gas() uint64 {
 // it automatically filled with the default amount, and the tx is broadcasted
 // again. Note that this may still end with the same error if the amount is
 // greater than the amount dumped by the faucet.
-func (s TxService) Broadcast() (Response, error) {
+func (s TxService) Broadcast(ctx context.Context) (Response, error) {
 	defer s.client.lockBech32Prefix()()
 
-	accountName := s.clientContext.GetFromName()
-	accountAddress := s.clientContext.GetFromAddress()
-
 	// validate msgs.
 	for _, msg := range s.txBuilder.GetTx().GetMsgs() {
 		if err := msg.ValidateBasic(); err != nil {
-			return Response{}, err
+			return Response{}, errors.WithStack(err)
 		}
 	}
 
+	accountName := s.clientContext.GetFromName()
 	if err := s.client.signer.Sign(s.txFactory, accountName, s.txBuilder, true); err != nil {
-		return Response{}, err
+		return Response{}, errors.WithStack(err)
 	}
 
 	txBytes, err := s.clientContext.TxConfig.TxEncoder()(s.txBuilder.GetTx())
 	if err != nil {
-		return Response{}, err
+		return Response{}, errors.WithStack(err)
 	}
 
 	resp, err := s.clientContext.BroadcastTx(txBytes)
-	if s.client.useFaucet && errors.Is(err, sdkerrors.ErrInsufficientFunds) {
-		err = s.client.makeSureAccountHasTokens(context.Background(), accountAddress.String())
-		if err != nil {
-			return Response{}, err
-		}
-		resp, err = s.clientContext.BroadcastTx(txBytes)
+	if err := handleBroadcastResult(resp, err); err != nil {
+		return Response{}, err
+	}
+
```

Comment:
Why did we remove the account balance check?

Reply:
The balance is already checked and eventually refunded in `CreateTx`. 

Because of that I never managed to write an integration test where this code is executed. I needed such an integration test to know if the error `ErrInsufficientFunds` is actually returned on `CheckTx` (so returned by `BroadcastTx`) or `DeliverTx` (so returned by `WaitForTx`). 

I went to the conclusion that this case is redundant, and removed it. But maybe you have something in mind ?